packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -166,64 +170,35 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
             throw error
         }
     }
-
-    private static func generateSerializedJsonData(values: [String: JSONValue], flutterModelRegistration: FlutterModels, modelName: String) throws -> [String: Any] {
-        let modelSchema = try getModelSchema(flutterModelRegistration: flutterModelRegistration, modelName: modelName)
-        var result = [String: Any]()
-
-        for (key, value) in values {
-            let field = modelSchema.field(withName: key)
-            if case .object(let deserializedValue) = value {
-                // If a field that has many models
-                if (deserializedValue["associatedField"] != nil && deserializedValue["associatedId"] != nil) {
-                    result[key] = Array<Any>();
-                }
-                // If a field that has one or belongs to a model
-                else if case .string(let modelId) = deserializedValue["id"],
-                        case .model(let nextModelName) = field!.type {
-                    result[key] = [
-                        "id": modelId,
-                        "modelName": nextModelName,
-                        "serializedData": try generateSerializedJsonData(values: deserializedValue, flutterModelRegistration: flutterModelRegistration, modelName: nextModelName)
-                    ]
-                }
-            } else if case .string(let deserializedValue) = value {
-                result[key] = deserializedValue
-            } else if case .boolean(let deserializedValue) = value {
-                result[key] = deserializedValue
-            } else if case .number(let deserializedValue) = value {
-                result[key] = deserializedValue
-            } else if case .null = value {
-                result[key] = nil
-            }
-        }
-
-        return result;
-    }
-
-    private func generateSerializedData(flutterModelRegistration: FlutterModels, modelName: String) throws -> [String: Any]{
+    
+    private func generateSerializedData(values: [String: JSONValue], flutterModelRegistration: FlutterModels, modelName: String) throws -> [String: Any]{
```

Comment:
Super untested but maybe we can make use of compound cases again:
```swift
private func generateSerializedData(values: [String: JSONValue], flutterModelRegistration: FlutterModels, modelName: String) throws -> [String: Any] {
    let modelSchema = try FlutterSerializedModel.getModelSchema(flutterModelRegistration: flutterModelRegistration, modelName: modelName)
    var result = [String: Any]()
    
    for(key, value) in values {
        let field = modelSchema.field(withName: key)
        
        switch (field?.type, value) {
            case (_, nil):
                continue
            case (_, .object(let deserializedValue)):
                // If a field that has many models
                if (deserializedValue["associatedField"] != nil && deserializedValue["associatedId"] != nil) {
                    result[key] = [Any]()
                } else {
                  fallthrough
                }
            case (.model(let nextModelName), .object(let deserializedValue)):
                // If a field that has one or belongs to a model
                if case .string(let modelId) = deserializedValue["id"] {
                    result[key] = [
                        "id": modelId,
                        "modelName": nextModelName,
                        "serializedData": try generateSerializedData(values: deserializedValue, flutterModelRegistration: flutterModelRegistration, modelName: nextModelName)
                    ]
                }
            case(.collection, _):
                continue
            case (.embeddedCollection(let fieldType, _), .array(let jsonArray)):
                result[key] = jsonArray.map {
                    deserializeValue(value: $0, fieldType: fieldType)
                }
            case (.dateTime, .some(.string(let deserializedValue))), (.date, .some(.string(let deserializedValue))), (.time, .some(.string(let deserializedValue))):
                result[key] = deserializedValue
            case (.timestamp, .some(.number(let deserializedValue))):
                result[key] = NSNumber(value: Int(deserializedValue))
            default:
                result[key] = extractJsonValue(key: key, value: value, modelSchema: modelSchema)!
        }
    }
    return result;
}
```
_Note_: Also updated `Array<Any>();` to `[Any]()` and used `.map` for the array case

Reply:
Thanks for the comments.  

Updated several methods to be static @HuiSF 

Added your code suggestions @cshfang

Reply:
Hey @cshfang thanks for the suggestions.  Unfortunately they add several compile errors.

1) 'fallthrough' from a case which doesn't bind variable 'nextModelName'

2) We use value and values[key] when checking the values.  I'm not sure how to add values[key] instead of value in the changes you suggested. 
![image](https://user-images.githubusercontent.com/11478656/130298956-1d858685-7aef-48b6-b735-c21976f0e332.png)

Personally I find the code changes make it a bit harder to read what's going on as well.

Reply:
Added the array and map changes you recommended though

Reply:
> 'fallthrough' from a case which doesn't bind variable 'nextModelName'

Darn, I didn't have my env set up properly to test this. It seems you can't just fallthrough without the associated values bound as the next case will skip the binding. Would have to rethink the logic a bit

> We use value and values[key] when checking the values. I'm not sure how to add values[key] instead of value in the changes you suggested.

Maybe I'm misreading this but isn't `value` and `values[key]` the same thing within the loop?


> Personally I find the code changes make it a bit harder to read what's going on as well.

It can certainly be kind of subjective and it seems like xcode might be doing some crazy formatting maybe.. I found it really difficult to parse through a bunch of nested cases. It's ok if it's too difficult to implement this as a single switch.